waku/v2/protocol/peer_exchange/waku_peer_exchange.go

```diff
@@ -122,6 +135,13 @@ func (wakuPX *WakuPeerExchange) handleResponse(ctx context.Context, response *pb
 
 	if len(peers) != 0 {
 		log.Info("connecting to newly discovered peers", zap.Int("count", len(peers)))
+
+		ch := make(chan peer.AddrInfo, len(peers))
+		for _, p := range peers {
+			ch <- p
+		}
+
+		wakuPX.connector.Connect(ctx, ch)
```

Comment:
again, it is likely I'm missing something:
here we connect to peers, but we do not remove any old code which would do the same thing. Did we not connect to peers before? It is likely a stupid question

Reply:
Oops. Indeed. I forgot to remove lines 145 to 154.
Good catch